In [122]:
import pandas as pd

In [123]:
rock_samples = pd.read_csv("data/rocksamples.csv")

In [124]:
rock_samples.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2229 entries, 0 to 2228
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   ID            2229 non-null   int64  
 1   Mission       2229 non-null   object 
 2   Type          2229 non-null   object 
 3   Subtype       2226 non-null   object 
 4   Weight (g)    2229 non-null   float64
 5   Pristine (%)  2229 non-null   float64
dtypes: float64(2), int64(1), object(3)
memory usage: 104.6+ KB


In [125]:
rock_samples.describe()

ID    Weight (g)  Pristine (%)
count   2229.000000   2229.000000   2229.000000
mean   52058.432032    168.253024     84.512764
std    26207.651471    637.286458     22.057299
min    10001.000000      0.000000      0.000000
25%    15437.000000      3.000000     80.010000
50%    65527.000000     10.200000     92.300000
75%    72142.000000     93.490000     98.140000
max    79537.000000  11729.000000    180.000000

In [126]:
rock_samples.head()

ID   Mission    Type   Subtype  Weight (g)  Pristine (%)
0  10001  Apollo11    Soil  Unsieved       125.8         88.36
1  10002  Apollo11    Soil  Unsieved      5629.0         93.73
2  10003  Apollo11  Basalt  Ilmenite       213.0         65.56
3  10004  Apollo11    Core  Unsieved        44.8         71.76
4  10005  Apollo11    Core  Unsieved        53.4         40.31

In [127]:
rock_samples.sample(9) #랜덤으로 샘플 뽑아줌

ID   Mission     Type     Subtype  Weight (g)  Pristine (%)
1443  68841  Apollo16     Soil        <1mm      266.60         93.27
1571  71048  Apollo17   Basalt    Ilmenite        2.46         73.55
565   15470  Apollo15     Soil    Unsieved       82.20         97.82
1586  71085  Apollo17   Basalt    Ilmenite        3.40         87.60
1933  76036  Apollo17  Breccia  ImpactMelt        3.95        100.00
129   12070  Apollo12     Soil    Unsieved     1102.00         80.24
725   60051  Apollo16     Soil        <1mm      195.30         89.25
210   14175  Apollo14  Breccia  Fragmental        7.48         92.91
125   12062  Apollo12   Basalt    Ilmenite      738.70         97.52

In [128]:
rock_samples.shape

(2229, 6)

In [129]:
rock_samples.columns

Index(['ID', 'Mission', 'Type', 'Subtype', 'Weight (g)', 'Pristine (%)'], dtype='object')

In [130]:
rock_samples.dtypes

ID                int64
Mission          object
Type             object
Subtype          object
Weight (g)      float64
Pristine (%)    float64
dtype: object

In [131]:
rock_samples.isnull().sum()

ID              0
Mission         0
Type            0
Subtype         3
Weight (g)      0
Pristine (%)    0
dtype: int64

In [132]:
rock_samples[rock_samples['Subtype'].isnull()]

ID   Mission     Type Subtype  Weight (g)  Pristine (%)
89    12023  Apollo12  Special     NaN      407.90         66.89
355   15014  Apollo15  Special     NaN      333.20        100.00
1513  70149  Apollo17   Basalt     NaN        0.95          0.00

In [133]:
rock_samples['Type'].unique()

array(['Soil', 'Basalt', 'Core', 'Breccia', 'Special', 'Crustal'],
      dtype=object)

In [134]:
rock_samples['Subtype'].unique()

array(['Unsieved', 'Ilmenite', 'Regolith', 'Cristobalite', 'Olivine',
       'Pigeonite', 'Fragmental', nan, 'Feldspathic', 'Unclassified',
       '2-4mm', '1-2mm', '<1mm', '4-10mm', '>10mm', 'ImpactMelt', 'KREEP',
       'Glass', 'Agglomerate', 'Unstudied', 'Cataclastic', 'Clod',
       'Anorthosite', 'unsieved', 'Plagioclase', 'fragmental',
       'unstudied', 'notstudied', 'Cataclasite', 'Troctolite', 'Norite',
       'Pristine'], dtype=object)

In [135]:
rock_samples['Subtype'].value_counts()

Subtype
ImpactMelt      393
Unsieved        271
Regolith        248
Fragmental      208
Ilmenite        184
<1mm            147
1-2mm           145
4-10mm          145
2-4mm           145
Olivine         115
Anorthosite      41
Pigeonite        37
Unclassified     24
fragmental       21
unsieved         17
Clod             17
Cataclasite      16
Agglomerate      10
unstudied        10
Feldspathic       8
Glass             7
Norite            4
KREEP             3
Cataclastic       3
Cristobalite      1
>10mm             1
Unstudied         1
notstudied        1
Plagioclase       1
Troctolite        1
Pristine          1
Name: count, dtype: int64

### type, subtype로 분류해보기
- size()는 보기 편하게 해줌

In [136]:
rock_samples.groupby(['Type', 'Subtype']).size()

Type     Subtype     
Basalt   Cristobalite      1
         Feldspathic       7
         Ilmenite        184
         KREEP             3
         Olivine         115
         Pigeonite        37
         Plagioclase       1
         Unclassified      2
Breccia  Agglomerate      10
         Anorthosite      21
         Clod             17
         Feldspathic       1
         Fragmental      207
         Glass             7
         ImpactMelt      393
         Regolith        248
         Unclassified     22
         Unstudied         1
         fragmental       21
         notstudied        1
         unstudied        10
Core     Unsieved         41
         unsieved         15
Crustal  Anorthosite      20
         Cataclasite      16
         Cataclastic       3
         Fragmental        1
         Norite            4
         Pristine          1
         Troctolite        1
Soil     1-2mm           145
         2-4mm           145
         4-10mm          145
         <1mm        

In [137]:
rock_samples['Weight (kg)'] = rock_samples['Weight (g)'] /1000
#rock_samples['Weight (g)'] = rock_samples['Weight (g)'].apply(lambda x: x * 0.001)
#rock.samples['Weight (g)'].rename(columns={'Weight (g)':'Weight (kg)'})

#### rock_samples.columns=[기존 df와 길이가 일치하는 열 이름 리스트 만들어서 넣기]

In [139]:
rock_samples

ID   Mission     Type   Subtype  Weight (g)  Pristine (%)  \
0     10001  Apollo11     Soil  Unsieved      125.80         88.36   
1     10002  Apollo11     Soil  Unsieved     5629.00         93.73   
2     10003  Apollo11   Basalt  Ilmenite      213.00         65.56   
3     10004  Apollo11     Core  Unsieved       44.80         71.76   
4     10005  Apollo11     Core  Unsieved       53.40         40.31   
...     ...       ...      ...       ...         ...           ...   
2224  79528  Apollo17  Breccia  Regolith        2.38        100.00   
2225  79529  Apollo17  Breccia  Regolith        1.84        100.00   
2226  79535  Apollo17  Breccia  Regolith        1.69        100.00   
2227  79536  Apollo17  Breccia  Regolith        1.66        100.00   
2228  79537  Apollo17  Breccia  Regolith        1.05        100.00   

      Weight (kg)  
0         0.12580  
1         5.62900  
2         0.21300  
3         0.04480  
4         0.05340  
...           ...  
2224      0.00238  
2225      0.00184  
2226      0.00169  
2227      0.00166  
2228      0.00105  

[2229 rows x 7 columns]

In [140]:
rock_samples.drop(columns='Weight (g)')

ID   Mission     Type   Subtype  Pristine (%)  Weight (kg)
0     10001  Apollo11     Soil  Unsieved         88.36      0.12580
1     10002  Apollo11     Soil  Unsieved         93.73      5.62900
2     10003  Apollo11   Basalt  Ilmenite         65.56      0.21300
3     10004  Apollo11     Core  Unsieved         71.76      0.04480
4     10005  Apollo11     Core  Unsieved         40.31      0.05340
...     ...       ...      ...       ...           ...          ...
2224  79528  Apollo17  Breccia  Regolith        100.00      0.00238
2225  79529  Apollo17  Breccia  Regolith        100.00      0.00184
2226  79535  Apollo17  Breccia  Regolith        100.00      0.00169
2227  79536  Apollo17  Breccia  Regolith        100.00      0.00166
2228  79537  Apollo17  Breccia  Regolith        100.00      0.00105

[2229 rows x 6 columns]

In [141]:
mission= pd.DataFrame()

In [142]:
mission["Mission"]=rock_samples["Mission"].unique()

In [143]:
mission

Mission
0  Apollo11
1  Apollo12
2  Apollo14
3  Apollo15
4  Apollo16
5  Apollo17

## 아폴로 임무별 샘플 중량 총합

In [144]:
a=rock_samples.groupby('Mission')['Weight (kg)'].sum()

In [145]:
type(a)

pandas.core.series.Series

In [146]:
a.values

array([ 21.55424,  34.34238,  41.83363,  75.3991 ,  92.46262, 109.44402])

In [147]:
mission['Weight (kg)'] = a.values

In [148]:
mission=mission.drop(columns='total_Weight(kg)')

KeyError: "['total_Weight(kg)'] not found in axis"

In [149]:
mission=mission.rename(columns={'Weight (kg)':'Sample_Weight (kg)'})

### 임무간 중량 차이 구하기

In [150]:
mission['Weight_diff']=mission['Sample_Weight (kg)'].diff()

In [151]:
misson=mission['Weight_diff'].fillna(value=0, inplace=True)

C:\Users\USER\AppData\Local\Temp\ipykernel_4336\4150101196.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  misson=mission['Weight_diff'].fillna(value=0, inplace=True)


In [152]:
mission

Mission  Sample_Weight (kg)  Weight_diff
0  Apollo11            21.55424      0.00000
1  Apollo12            34.34238     12.78814
2  Apollo14            41.83363      7.49125
3  Apollo15            75.39910     33.56547
4  Apollo16            92.46262     17.06352
5  Apollo17           109.44402     16.98140

### 달 모듈의 이름, 중량 컬럼을 mission에 추가
- 달 모듈과 명령 모듈이 있고 이 두가지를 일컫어 승무원 어쩌구라고 한다.

In [153]:
달모듈이름 = ['Eagle (LM-5)', 'Intrepid (LM-6)', 'Antares (LM-8)', 'Falcon (LM-10)', 'Orion (LM-11)', 'Challenger (LM-12)']
달모듈중량 = [15103, 15235, 15264, 16430, 16445, 16456]

In [154]:
mission['Lunar module (LM)'] = 달모듈이름
mission['LM mass (kg)'] = 달모듈중량
mission

Mission  Sample_Weight (kg)  Weight_diff   Lunar module (LM)  LM mass (kg)
0  Apollo11            21.55424      0.00000        Eagle (LM-5)         15103
1  Apollo12            34.34238     12.78814     Intrepid (LM-6)         15235
2  Apollo14            41.83363      7.49125      Antares (LM-8)         15264
3  Apollo15            75.39910     33.56547      Falcon (LM-10)         16430
4  Apollo16            92.46262     17.06352       Orion (LM-11)         16445
5  Apollo17           109.44402     16.98140  Challenger (LM-12)         16456

In [164]:
mission['LM mass diff']=mission['LM mass (kg)'].diff()

In [166]:
mission['LM mass diff'].fillna(value=0, inplace=True)

C:\Users\USER\AppData\Local\Temp\ipykernel_4336\1992362754.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  mission['LM mass diff'].fillna(value=0, inplace=True)


In [167]:
mission

Mission  Sample_Weight (kg)  Weight_diff   Lunar module (LM)  \
0  Apollo11            21.55424      0.00000        Eagle (LM-5)   
1  Apollo12            34.34238     12.78814     Intrepid (LM-6)   
2  Apollo14            41.83363      7.49125      Antares (LM-8)   
3  Apollo15            75.39910     33.56547      Falcon (LM-10)   
4  Apollo16            92.46262     17.06352       Orion (LM-11)   
5  Apollo17           109.44402     16.98140  Challenger (LM-12)   

   LM mass (kg)  LM mass diff  
0         15103           0.0  
1         15235         132.0  
2         15264          29.0  
3         16430        1166.0  
4         16445          15.0  
5         16456          11.0

In [180]:
명령모듈이름 = ['Columbia (CSM-107]', 'Yankee Clipper (CM-100])', 'Kitty Hawk (CM-110)', 'Endeavor (CM-112)', 'Casper (CM-113)', 'America (CM-114)']
명령모듈중량 = [5560, 5609, 5758, 5875, 5840, 5960]

In [181]:
mission['Command Module (CM)'] = 명령모듈이름
mission['CM Mass (kg)'] = 명령모듈중량

In [182]:
mission['CM Mass diff']=mission['CM Mass (kg)'].diff()

In [184]:
mission['CM Mass diff'].fillna(value = 0, inplace=True)

C:\Users\USER\AppData\Local\Temp\ipykernel_4336\2128624598.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  mission['CM Mass diff'].fillna(value = 0, inplace=True)


In [ ]:
mission['Total weight (kg)']=mission['LM mass (kg)']+mission['CM Mass (kg)']

In [ ]:
mission['Total weight diff']= mission['Total weight (kg)'].diff()
mission['Total weight diff'].fillna(value=0, inplace=True)

C:\Users\USER\AppData\Local\Temp\ipykernel_4336\3435717993.py:2: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  mission['Total weight diff'].fillna(value=0, inplace=True)


In [193]:
mission

Mission  Sample_Weight (kg)  Weight_diff   Lunar module (LM)  \
0  Apollo11            21.55424      0.00000        Eagle (LM-5)   
1  Apollo12            34.34238     12.78814     Intrepid (LM-6)   
2  Apollo14            41.83363      7.49125      Antares (LM-8)   
3  Apollo15            75.39910     33.56547      Falcon (LM-10)   
4  Apollo16            92.46262     17.06352       Orion (LM-11)   
5  Apollo17           109.44402     16.98140  Challenger (LM-12)   

   LM mass (kg)  LM mass diff       Command Module (CM)  CM Mass (kg)  \
0         15103           0.0        Columbia (CSM-107]          5560   
1         15235         132.0  Yankee Clipper (CM-100])          5609   
2         15264          29.0       Kitty Hawk (CM-110)          5758   
3         16430        1166.0         Endeavor (CM-112)          5875   
4         16445          15.0           Casper (CM-113)          5840   
5         16456          11.0          America (CM-114)          5960   

   Total weight (kg)  CM Mass diff  Total weight diff  
0              20663           0.0                0.0  
1              20844          49.0              181.0  
2              21022         149.0              178.0  
3              22305         117.0             1283.0  
4              22285         -35.0              -20.0  
5              22416         120.0              131.0

In [ ]:
saturnVPayload = 435000

# 페이로드에서 승무원 영역이 차지하는 비율
mission['Crewed area : Payload'] = mission['Total weight (kg)']/saturnVPayload
mission

Mission  Sample_Weight (kg)  Weight_diff   Lunar module (LM)  \
0  Apollo11            21.55424      0.00000        Eagle (LM-5)   
1  Apollo12            34.34238     12.78814     Intrepid (LM-6)   
2  Apollo14            41.83363      7.49125      Antares (LM-8)   
3  Apollo15            75.39910     33.56547      Falcon (LM-10)   
4  Apollo16            92.46262     17.06352       Orion (LM-11)   
5  Apollo17           109.44402     16.98140  Challenger (LM-12)   

   LM mass (kg)  LM mass diff       Command Module (CM)  CM Mass (kg)  \
0         15103           0.0        Columbia (CSM-107]          5560   
1         15235         132.0  Yankee Clipper (CM-100])          5609   
2         15264          29.0       Kitty Hawk (CM-110)          5758   
3         16430        1166.0         Endeavor (CM-112)          5875   
4         16445          15.0           Casper (CM-113)          5840   
5         16456          11.0          America (CM-114)          5960   

   Total weight (kg)  CM Mass diff  Total weight diff  Crewed area : Payload  
0              20663           0.0                0.0               0.047501  
1              20844          49.0              181.0               0.047917  
2              21022         149.0              178.0               0.048326  
3              22305         117.0             1283.0               0.051276  
4              22285         -35.0              -20.0               0.051230  
5              22416         120.0              131.0               0.051531

In [ ]:
# 승무원 영역에서 암석 샘플이 차지하는 비율
temp = mission['Sample_Weight (kg)'] / mission['Total weight (kg)']
temp

0    0.001043
1    0.001648
2    0.001990
3    0.003380
4    0.004149
5    0.004882
dtype: float64

In [206]:
mission['Sample : Crewed area'] = temp
mission

Mission  Sample_Weight (kg)  Weight_diff   Lunar module (LM)  \
0  Apollo11            21.55424      0.00000        Eagle (LM-5)   
1  Apollo12            34.34238     12.78814     Intrepid (LM-6)   
2  Apollo14            41.83363      7.49125      Antares (LM-8)   
3  Apollo15            75.39910     33.56547      Falcon (LM-10)   
4  Apollo16            92.46262     17.06352       Orion (LM-11)   
5  Apollo17           109.44402     16.98140  Challenger (LM-12)   

   LM mass (kg)  LM mass diff       Command Module (CM)  CM Mass (kg)  \
0         15103           0.0        Columbia (CSM-107]          5560   
1         15235         132.0  Yankee Clipper (CM-100])          5609   
2         15264          29.0       Kitty Hawk (CM-110)          5758   
3         16430        1166.0         Endeavor (CM-112)          5875   
4         16445          15.0           Casper (CM-113)          5840   
5         16456          11.0          America (CM-114)          5960   

   Total weight (kg)  CM Mass diff  Total weight diff  Crewed area : Payload  \
0              20663           0.0                0.0               0.047501   
1              20844          49.0              181.0               0.047917   
2              21022         149.0              178.0               0.048326   
3              22305         117.0             1283.0               0.051276   
4              22285         -35.0              -20.0               0.051230   
5              22416         120.0              131.0               0.051531   

   Sample : Crewed area  
0              0.001043  
1              0.001648  
2              0.001990  
3              0.003380  
4              0.004149  
5              0.004882

In [ ]:
# 페이로드에서 샘플이 차지하는 비율
mission['Sample : Payload'] = mission['Sample_Weight (kg)'] / saturnVPayload

In [209]:
mission

Mission  Sample_Weight (kg)  Weight_diff   Lunar module (LM)  \
0  Apollo11            21.55424      0.00000        Eagle (LM-5)   
1  Apollo12            34.34238     12.78814     Intrepid (LM-6)   
2  Apollo14            41.83363      7.49125      Antares (LM-8)   
3  Apollo15            75.39910     33.56547      Falcon (LM-10)   
4  Apollo16            92.46262     17.06352       Orion (LM-11)   
5  Apollo17           109.44402     16.98140  Challenger (LM-12)   

   LM mass (kg)  LM mass diff       Command Module (CM)  CM Mass (kg)  \
0         15103           0.0        Columbia (CSM-107]          5560   
1         15235         132.0  Yankee Clipper (CM-100])          5609   
2         15264          29.0       Kitty Hawk (CM-110)          5758   
3         16430        1166.0         Endeavor (CM-112)          5875   
4         16445          15.0           Casper (CM-113)          5840   
5         16456          11.0          America (CM-114)          5960   

   Total weight (kg)  CM Mass diff  Total weight diff  Crewed area : Payload  \
0              20663           0.0                0.0               0.047501   
1              20844          49.0              181.0               0.047917   
2              21022         149.0              178.0               0.048326   
3              22305         117.0             1283.0               0.051276   
4              22285         -35.0              -20.0               0.051230   
5              22416         120.0              131.0               0.051531   

   Sample : Crewed area  Sample : Payload  
0              0.001043          0.000050  
1              0.001648          0.000079  
2              0.001990          0.000096  
3              0.003380          0.000173  
4              0.004149          0.000213  
5              0.004882          0.000252

### 아르테미스 미션 프레임 만들기

In [ ]:
artemis_crewedArea = 26520

artemis_dict = {'Mission' : ['artemis1', 'artemis1b', 'artemis2'],
                'Total weight (kg)' : [artemis_crewedArea, artemis_crewedArea, artemis_crewedArea],
                'Payload (kg)' : [26988, 37965, 42955]
               }

#딕셔너리를 데이터 프레임으로 저장하기
artemis_mission = pd.DataFrame(artemis_dict)

In [213]:
artemis_mission

Mission  Total weight (kg)  Payload (kg)
0   artemis1              26520         26988
1  artemis1b              26520         37965
2   artemis2              26520         42955

In [214]:
cp=mission['Crewed area : Payload'].mean()
sp=mission['Sample : Payload'].mean()
sc=mission['Sample : Crewed area'].mean()

In [217]:
cp, sp, sc

(np.float64(0.04963026819923372),
 np.float64(0.00014369195019157087),
 np.float64(0.002848764392685611))

In [236]:
artemis_mission['Sample weight from total(kg)'] = artemis_mission['Total weight (kg)'] * sc
artemis_mission['Sample weight from payload(kg)'] = artemis_mission['Payload (kg)']*10 * sp

In [237]:
artemis_mission['Estimated sample weight'] =  (artemis_mission['Sample weight from payload(kg)']+artemis_mission['Sample weight from total(kg)'])/2

In [242]:
artemis_mission

Mission  Total weight (kg)  Payload (kg)  Sample weight from totaln(kg)  \
0   artemis1              26520         26988                      75.549232   
1  artemis1b              26520         37965                      75.549232   
2   artemis2              26520         42955                      75.549232   

   Sample weight from payload(kg)  Sample weight from total(kg)  \
0                       38.779584                     75.549232   
1                       54.552649                     75.549232   
2                       61.722877                     75.549232   

   Estimated sample weight  
0                57.164408  
1                65.050940  
2                68.636054

## 아르테미스 임무에서 가져올 암석 종류 구하기

In [244]:
rock_samples.head()

ID   Mission    Type   Subtype  Weight (g)  Pristine (%)  Weight (kg)
0  10001  Apollo11    Soil  Unsieved       125.8         88.36       0.1258
1  10002  Apollo11    Soil  Unsieved      5629.0         93.73       5.6290
2  10003  Apollo11  Basalt  Ilmenite       213.0         65.56       0.2130
3  10004  Apollo11    Core  Unsieved        44.8         71.76       0.0448
4  10005  Apollo11    Core  Unsieved        53.4         40.31       0.0534

In [251]:
rock_samples['Remaining (kg)']=(rock_samples['Weight (kg)']*rock_samples['Pristine (%)'])*0.001

In [265]:
rock_samples.drop(columns='Weight (g)', inplace = True)

In [266]:
rock_samples

ID   Mission     Type   Subtype  Pristine (%)  Weight (kg)  \
0     10001  Apollo11     Soil  Unsieved         88.36      0.12580   
1     10002  Apollo11     Soil  Unsieved         93.73      5.62900   
2     10003  Apollo11   Basalt  Ilmenite         65.56      0.21300   
3     10004  Apollo11     Core  Unsieved         71.76      0.04480   
4     10005  Apollo11     Core  Unsieved         40.31      0.05340   
...     ...       ...      ...       ...           ...          ...   
2224  79528  Apollo17  Breccia  Regolith        100.00      0.00238   
2225  79529  Apollo17  Breccia  Regolith        100.00      0.00184   
2226  79535  Apollo17  Breccia  Regolith        100.00      0.00169   
2227  79536  Apollo17  Breccia  Regolith        100.00      0.00166   
2228  79537  Apollo17  Breccia  Regolith        100.00      0.00105   

      Remaining (kg)  
0           0.011116  
1           0.527606  
2           0.013964  
3           0.003215  
4           0.002153  
...              ...  
2224        0.000238  
2225        0.000184  
2226        0.000169  
2227        0.000166  
2228        0.000105  

[2229 rows x 7 columns]

In [285]:
low_samples = rock_samples[(rock_samples['Weight (kg)'] >= 0.16) & (rock_samples['Pristine (%)'] <= 50)]

In [286]:
low_samples.info()

<class 'pandas.core.frame.DataFrame'>
Index: 27 entries, 11 to 2183
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   ID              27 non-null     int64  
 1   Mission         27 non-null     object 
 2   Type            27 non-null     object 
 3   Subtype         27 non-null     object 
 4   Pristine (%)    27 non-null     float64
 5   Weight (kg)     27 non-null     float64
 6   Remaining (kg)  27 non-null     float64
dtypes: float64(3), int64(1), object(3)
memory usage: 1.7+ KB


In [287]:
low_samples['Type'].value_counts()

Type
Basalt     14
Breccia     8
Soil        4
Core        1
Name: count, dtype: int64

In [288]:
needed_sample=low_samples[low_samples['Type'].isin(['Basalt', 'Breccia'])]

In [289]:
needed_sample

ID   Mission     Type      Subtype  Pristine (%)  Weight (kg)  \
11    10017  Apollo11   Basalt     Ilmenite         43.71       0.9730   
14    10020  Apollo11   Basalt     Ilmenite         27.88       0.4250   
15    10021  Apollo11  Breccia     Regolith         30.21       0.2500   
29    10045  Apollo11   Basalt      Olivine         12.13       0.1850   
37    10057  Apollo11   Basalt     Ilmenite         35.15       0.9190   
39    10059  Apollo11  Breccia     Regolith         36.94       0.1880   
52    10072  Apollo11   Basalt     Ilmenite         15.22       0.4470   
68    12002  Apollo12   Basalt      Olivine         49.04       1.5300   
72    12006  Apollo12   Basalt      Olivine          0.53       0.2064   
113   12047  Apollo12   Basalt     Ilmenite         33.70       0.1930   
118   12053  Apollo12   Basalt    Pigeonite         46.43       0.8790   
120   12055  Apollo12   Basalt    Pigeonite         26.00       0.9120   
426   15205  Apollo15  Breccia     Regolith         24.70       0.3373   
534   15405  Apollo15  Breccia   ImpactMelt         35.91       0.5131   
570   15475  Apollo15   Basalt    Pigeonite         49.60       0.4068   
575   15498  Apollo15  Breccia     Regolith         34.11       2.3400   
607   15555  Apollo15   Basalt      Olivine         46.61       9.6140   
718   60016  Apollo16  Breccia   Fragmental         49.57       4.3070   
722   60025  Apollo16  Breccia  Anorthosite         43.61       1.8360   
1221  66075  Apollo16  Breccia   Fragmental         32.97       0.3471   
1645  71566  Apollo17   Basalt     Ilmenite         45.06       0.4144   
2183  79155  Apollo17   Basalt     Ilmenite         25.97       0.3188   

      Remaining (kg)  
11          0.042530  
14          0.011849  
15          0.007553  
29          0.002244  
37          0.032303  
39          0.006945  
52          0.006803  
68          0.075031  
72          0.000109  
113         0.006504  
118         0.040812  
120         0.023712  
426         0.008331  
534         0.018425  
570         0.020177  
575         0.079817  
607         0.448109  
718         0.213498  
722         0.080068  
1221        0.011444  
1645        0.018673  
2183        0.008279

In [290]:
rock_samples['Type'].value_counts()

Type
Breccia    959
Soil       813
Basalt     351
Core        56
Crustal     46
Special      4
Name: count, dtype: int64

In [296]:
crustal = rock_samples[rock_samples['Type']=='Crustal']

In [301]:
needed_sample = pd.concat([needed_sample, crustal])

### neeeded samples overview 만들기
- needed samples : 어떤 암석 종류를 체취하겠냐를 정함
- basalt, breccia, crustal

In [313]:
overview = needed_sample.groupby('Type')["Weight (kg)"].sum().reset_index()

In [314]:
overview

Type  Weight (kg)
0   Basalt     17.42340
1  Breccia     10.11850
2  Crustal      4.74469

In [316]:
overview.rename(columns = {'Weight (kg)':'Total weight (kg)'},inplace = True)

In [323]:
# reset.index() -> 인덱스 초기화해줌
temp = needed_sample.groupby('Type')["Weight (kg)"].mean().reset_index()

Type  Weight (kg)
0   Basalt     1.244529
1  Breccia     1.264813
2  Crustal     0.103145

In [325]:
overview= pd.merge(overview, temp, on = 'Type')

In [326]:
overview

Type  Total weight (kg)  Weight (kg)
0   Basalt           17.42340     1.244529
1  Breccia           10.11850     1.264813
2  Crustal            4.74469     0.103145

In [327]:
overview.rename(columns = {'Weight (kg)':'Average weight (kg)'},inplace = True)

In [328]:
overview

Type  Total weight (kg)  Average weight (kg)
0   Basalt           17.42340             1.244529
1  Breccia           10.11850             1.264813
2  Crustal            4.74469             0.103145

In [ ]:
temp = rock_samples.groupby('Type')['ID'].count().reset_index()
# ID는 아무런 영향을 미치지 않기 때문에 넣은거임(그냥 타입 별로 수를 세야해서)

In [335]:
overview= pd.merge(overview, temp, on='Type')

In [338]:
overview

Type  Total weight (kg)  Average weight (kg)   ID
0   Basalt           17.42340             1.244529  351
1  Breccia           10.11850             1.264813  959
2  Crustal            4.74469             0.103145   46

In [343]:
overview.rename(columns={'ID':'Number of samples'}, inplace=True)

In [344]:
overview

Type  Total weight (kg)  Average weight (kg)  Number of samples
0   Basalt           17.42340             1.244529                351
1  Breccia           10.11850             1.264813                959
2  Crustal            4.74469             0.103145                 46

In [356]:
temp = overview['Number of samples'].sum()

In [357]:
overview['Percentage']=overview["Number of samples"]/temp

In [360]:
overview

Type  Total weight (kg)  Average weight (kg)  Number of samples  \
0   Basalt           17.42340             1.244529                351   
1  Breccia           10.11850             1.264813                959   
2  Crustal            4.74469             0.103145                 46   

   Percentage  
0    0.258850  
1    0.707227  
2    0.033923

## 수집할 암석 별 무게

In [361]:
temp=artemis_mission['Estimated sample weight'].mean()

In [362]:
overview['Weight to Collect']=overview['Percentage']*temp

In [363]:
overview

Type  Total weight (kg)  Average weight (kg)  Number of samples  \
0   Basalt           17.42340             1.244529                351   
1  Breccia           10.11850             1.264813                959   
2  Crustal            4.74469             0.103145                 46   

   Percentage  Weight to Collect  
0    0.258850          16.467267  
1    0.707227          44.991764  
2    0.033923           2.158103

## 수집할 암석 별 개수

In [365]:
overview['Rock to collect'] = overview['Weight to Collect']/overview['Average weight (kg)']

In [366]:
overview

Type  Total weight (kg)  Average weight (kg)  Number of samples  \
0   Basalt           17.42340             1.244529                351   
1  Breccia           10.11850             1.264813                959   
2  Crustal            4.74469             0.103145                 46   

   Percentage  Weight to Collect  Rock to collect  
0    0.258850          16.467267        13.231731  
1    0.707227          44.991764        35.571884  
2    0.033923           2.158103        20.922917

## 결론

- 암석을 실을 수 있는 최대 무게는 약 63kg이므로 암석 별 무게를 고려하여 최대 무게를 넘어가지 않도록 암석 별로 채굴할 개수를 구했다.  
    - 암석 채굴 비율
        - Basalt: 25.9%
        - Breccia: 70.7%
        - Crustal: 2.4%
    - 암석 평균 무게
        - Basalt: 1.25kg
        - Breccia: 1.26kg
        - Crustal: 0.1kg
    - 암석 채굴 개수
        - Basalt: 13개
        - Breccia: 36개
        - Crustal: 21개
- Basalt와 Breccia는 채굴량이 많고, 사용량 또한 많기 때문에 다음 미션에서 더 채굴해와야할 필요가 있다.
- Crustal 같은 경우에는 채굴량 자체는 적으나 그 이유가 희귀하기 때문이고 사용량 또한 적지 않으므로 다음 미션에서 채굴해와야한다.
